flight/search/plot_algo.py

```diff
@@ -0,0 +1,601 @@
+"""
+The actual algorithm powering the drone's navigation
+"""
+
+from math import floor, sqrt
+from bisect import insort
+from pathfinding.finder.a_star import AStarFinder
+from pathfinding.core.grid import Grid, Node
+from numpy import zeros, int8
+from numpy.typing import NDArray
```

Comment:
The standard way to import numpy is to do `import numpy as np`. Normally it would be fine to just import the 2 things that you need, but vision will already be importing numpy, so you should import the whole library.

Also, numpy typing should be done with the `nptyping` module, not `numpy.typing`. They are very similar, but `numpy.typing` is more advanced, so we are using that. If you need examples on how to do this, check out [https://missourimrr.github.io/docs/python/type-annotations/full-guide.html#numpy-arrays](https://missourimrr.github.io/docs/python/type-annotations/full-guide.html#numpy-arrays)